systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/ListenersST.java

```diff
@@ -2006,6 +2010,112 @@ void testCertificateWithNonExistingDataKey(ExtensionContext extensionContext) {
+            .addToData("password", firstEncodedPassword)
+            .build();
+
+        kubeClient().namespace(namespaceName).createSecret(password);
+        assertThat("Password in secret is not correct", kubeClient().namespace(namespaceName).getSecret(secretName).getData().get("password"), is(firstEncodedPassword));
+
+        KafkaUser kafkaUser = KafkaUserTemplates.scramShaUser(clusterName, userName)
+            .editSpec()
+                .withNewKafkaUserScramSha512ClientAuthentication()
+                    .withNewPassword()
+                        .withNewValueFrom()
+                            .withNewSecretKeyRef("password", secretName, false)
+                        .endValueFrom()
+                    .endPassword()
+                .endKafkaUserScramSha512ClientAuthentication()
+            .endSpec()
+            .build();
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 3)
+            .editSpec()
+                .editKafka()
+                    .withListeners(new GenericKafkaListenerBuilder()
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                        .withPort(9096)
+                        .withTls(true)
+                        .withNewKafkaListenerAuthenticationScramSha512Auth()
+                        .endKafkaListenerAuthenticationScramSha512Auth()
+                        .build())
+                .endKafka()
+            .endSpec()
+            .build(),
+            kafkaUser,
+            KafkaTopicTemplates.topic(clusterName, topicName).build()
+        );
+
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(namespaceName, true, kafkaClientsName, kafkaUser).build());
+
+        String kafkaClientsPodName =
+            kubeClient(namespaceName).listPodsByPrefixInName(namespaceName, kafkaClientsName).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(kafkaClientsPodName)
+            .withTopicName(topicName)
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withKafkaUsername(userName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        int sentMessages = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(
+            sentMessages,
+            internalKafkaClient.receiveMessagesTls()
+        );
+
+        LOGGER.info("Changing password in secret: {}, we should be able to send/receive messages", secretName);
+
+        password = new SecretBuilder(password)
+            .addToData("password", secondEncodedPassword)
+            .build();
+
+        kubeClient().namespace(namespaceName).createSecret(password);
```

Comment:
Would something like replace make more sense?

Reply:
The `..createSecret(...)` method is just encapsulated `client.secrets().inNamespace(getNamespace()).createOrReplace(secret);` so I guess rename it to `createOrReplaceSecret(..)` or something like that will be enough?

Reply:
Ok. I guess for another PR and not high prio ... but it might make more sense to call the method what it is and not what it isn't.

Reply:
Yeah, I saw there some other methods, which should be renamed. I'll do it in another PR. Good catch! :)